### PR TITLE
Fixes in mlCompared.html

### PR DESCRIPTION
**Pull request by rizo**
Many small fixes: added semicolons, removed duplicate code, adjusted indentation and switched some table columns for consistency with the rest of the document.

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by chenglou** [comment]
Oh man this page needed so many changes. Thanks! Will merge after the CLA.

**Comment by rizo** [comment]
Signed the CLA, should I update the PR?

**Comment by facebook-github-bot-2** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: mlCompared.html [hunk 1]
```diff
@@ -326,25 +326,15 @@
 let myFuncs = {
   myFun = (fun x -> x + 1);
   your = (fun a b -> a + b);
-}</pre>
+};</pre>
```
Comment (chenglou) [inline mlCompared.html]: This is ocaml code, so no semicolon is right

File: mlCompared.html [hunk 2]
```diff
@@ -326,25 +326,15 @@
...
-}</pre>
-    </td>
-  <tr>
-    <td>
-      <pre>
-let myRec = {x = 0; y = 10}</pre>
-    </td>
-    <td>
-      <pre>
-let myRec = {x: 0, y: 10};</pre>
```
Comment (chenglou) [inline mlCompared.html]: why is this removed?

File: mlCompared.html [hunk 3]
```diff
@@ -711,13 +704,41 @@
   *is* a tuple.
 
 <table>
-  <thead><tr> <th scope="col"><p>Reason</p></th><th scope="col"><p>OCaml</p></th></tr></thead>
+  <thead><tr> <th scope="col"><p>OCaml</p></th><th scope="col"><p>Reason</p></th></tr></thead>
```
Comment (chenglou) [inline mlCompared.html]: Aaaah, thanks =).

File: mlCompared.html (hunk 2 above)
Comment (rizo) [inline mlCompared.html]: Because it's a duplicate, see lines [L316](https://github.com/facebook/reason/blob/907c5d3b3ba0011cc9fbedf1372dbb38e62c23c9/mlCompared.html#L316) and [L341](https://github.com/facebook/reason/blob/907c5d3b3ba0011cc9fbedf1372dbb38e62c23c9/mlCompared.html#L341).

File: mlCompared.html (hunk 1 above)
Comment (rizo) [inline mlCompared.html]: Right, my mistake.

**Comment by chenglou** [comment]
Cool. Thanks so much!

File: mlCompared.html [hunk 4]
```diff
@@ -664,17 +657,17 @@
 The following examples shows the difference between passing *two* type
 parameters to `list`, and a *single* type parameter that happens to be a tuple.
 <table>
-  <thead><tr> <th scope="col"><p>Reason</p></th><th scope="col"><p>OCaml</p></th></tr></thead>
+  <thead><tr> <th scope="col"><p>OCaml</p></th><th scope="col"><p>Reason</p></th></tr></thead>
```
Comment (jordwalke) [inline mlCompared.html]: Thank you!

**Comment by jordwalke** [comment]
Thanks @rizo!
